app/src/org/commcare/android/view/GraphView.java

```diff
@@ -181,8 +196,26 @@ private void renderSeries(SeriesData s) throws InvalidStateException {
         for (XYPointData d : s.getPoints()) {
             sortedPoints.add(d);
         }
-        Collections.sort(sortedPoints, new PointComparator());
+        Comparator<XYPointData> comparator;
+        if (mData.getType().equals(Graph.TYPE_BAR)) {
+            String barSort = s.getConfiguration("bar-sort", "");
+            if (barSort.equals("ascending")) {
+                comparator = new AscendingValuePointComparator();
+            }
+            else if (barSort.equals("descending")) {

```

Comment:
Style-wise _I think_ we've settled with putting 'else' on the same line as closing brace of the 'if'